tests/test_commons_auth.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+import logging
+import sys
+import unittest
+from contextlib import contextmanager
+
+import connexion.apis.abstract
+import os
+import requests
+
+pkg_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))  # noqa
+sys.path.insert(0, pkg_root)  # noqa
+
+import dss
+from dss.logging import configure_test_logging
+from dss.util import UrlBuilder
+from tests import get_auth_header
+from tests.infra.server import ThreadedLocalServer
+from dss.config import override_bucket_config, BucketConfig, Replica
+from tests.infra import DSSAssertMixin, DSSUploadMixin, get_env, testmode
+
+logger = logging.getLogger(__name__)
+
+
```

Comment:
It is also worth adding a note that this test suite could be made more comprehensive and more resilient to API changes by loading the swagger spec and using that to identify and test that all API methods require valid authentication.

Reply:
Added.